pkg/skip_response_writer.go

```diff
@@ -0,0 +1,69 @@
+package goa
+
+import (
+	"io"
+	"sync"
+	"sync/atomic"
+)
+
+// SkipResponseWriter converts an io.WriterTo into a io.ReadCloser.
+// The Read/Close methods this function returns will pipe the Write calls that wt makes, to implement a Reader that has the written bytes.
+// If Read is called Close must also be called to avoid leaking memory.
+// The returned value implements io.WriterTo as well, so the generated handler will call that instead of the Read method.
+//
+// Server handlers that use SkipResponseBodyEncodeDecode() io.ReadCloser as a return type.
+func SkipResponseWriter(wt io.WriterTo) io.ReadCloser {
+	return &writerToReaderAdapter{WriterTo: wt}
+}
+
+type writerToReaderAdapter struct {
+	io.WriterTo
+	prOnce sync.Once
+	pr     *io.PipeReader
+}
+
+func (a *writerToReaderAdapter) initPipe() {
+	if a.pr != nil {
```

Comment:
Is this test necessary?

Reply:
Nope, that's a holdover from before switching to sync.Once.